### PR TITLE
C++11 cleanup on constructors dir network

### PR DESCRIPTION
**Pull request by Dumbeldor**

### CODE REVIEW

File: src/network/connection.cpp [hunk 1]
```diff
@@ -2031,8 +1981,7 @@ void ConnectionSendThread::sendAsPacket(u16 peer_id, u8 channelnum,
 }
 
 ConnectionReceiveThread::ConnectionReceiveThread(unsigned int max_packet_size) :
```
Comment (Dumbeldor) [inline src/network/connection.cpp]: max_packet_size isn't used, I remove it or can it be used in the near future?

File: src/network/connection.h [hunk 2]
```diff
@@ -625,7 +628,7 @@ class PeerHelper
 	bool          operator!=(void* ptr);
 
 private:
-	Peer* m_peer;
+	Peer* m_peer = 0;
```
Comment (nerzhul) [inline src/network/connection.h]: nullptr & check if NULL is tested on this variable

File: src/network/connection.h [hunk 3]
```diff
@@ -758,16 +748,21 @@ class Peer {
 			float min_rtt;
 			float max_rtt;
 			float avg_rtt;
+
+			rttstats(float jitter_min_ = FLT_MAX, float jitter_max_ = 0.0f, float jitter_avg_ = -1.0f,
```
Comment (nerzhul) [inline src/network/connection.h]: what is that ?

File: src/network/connection.cpp [hunk 4]
```diff
@@ -812,7 +812,7 @@ void Channel::UpdateTimers(float dtime,bool legacy_peer)
 PeerHelper::PeerHelper(Peer* peer) :
 	m_peer(peer)
 {
-	if (peer != NULL)
+	if (peer)
 	{
```
Comment (nerzhul) [inline src/network/connection.cpp]: please fix braces too

**Comment by Zeno-** [comment]
:+1: pending braces being fixed

**Pull request merged**